### ISSUE TITLE
Can't run odata2ts CLI within package type module (require() of ES modules is not supported.)

### ISSUE THREAD

**Issue by manuelseeger** [odata2ts/odata2ts#152]
When trying to run the generator CLI in a package of "type": "module" I am getting the following error: 

```
> node node_modules\@odata2ts\odata2ts\lib\run-cli.js
Bad arguments! TypeScriptLoader failed to compile TypeScript:
Must use import to load ES Module: .\odata2ts.config.ts
require() of ES modules is not supported.
require() of .\odata2ts.config.ts from .\node_modules\cosmiconfig-typescript-loader\dist\cjs\index.js is an ES module file as it is a .ts file whose nearest parent package.json contains "type": "module" which defines all .ts files in that package scope as ES modules.
Instead change the requiring code to use import(), or remove "type": "module" from .\package.json.
```

It appears that this is a limitation of cosmiconfic-typescript-loader. See https://github.com/cosmiconfig/cosmiconfig/pull/283

Not sure if odata2ts can work around this. In any case leaving the issue here so it can be solved either by an updated cosmiconfig or a workaround.

**Comment by texttechne** [comment]
@manuelseeger This has been reported to be fixed with `cosmiconfig@8.2.0`. 
Please try `@odata2ts/odata2ts@0.24.0` if it solves the problem.

**Comment by texttechne** [comment]
@manuelseeger ping!

**Comment by manuelseeger** [comment]
I gave it a try with @odata2ts/odata2ts@0.25.1 and cosmiconfig@8.2.0 but I am still getting the same error. 

Not sure if I am doing something wrong, can you reproduce this?

**Comment by texttechne** [comment]
Nope. Can you give me an example of your setup?

**Comment by texttechne** [comment]
I've never used ESM, hence thanks for the example.
So the only thing to do is to add `type: "module"` in the `package.json` (as you mentioned in the beginning :see_no_evil:)...

Currently it seems that `cosmiconfig-typescript-loader` needs to work out [ESM support](https://github.com/Codex-/cosmiconfig-typescript-loader/issues/97).

I've never dived into the topic: What would I need to do for odata2ts to support this. I mean, I only have TS source files and they get compiled to commonjs, but how would I support both commonjs and ESM? Is this an either-or choice?

**Comment by texttechne** [comment]
Hi @manuelseeger,

so I've also updated `cosmiconfig-typescript-loader` to the newest version and it seems to work.

My minimal test case is the trippin example package: I've added `type: module` to the `package.json` and the generation process works fine after the upgrade: no more errors.

Released with `@odata2ts/odata2ts` version `0.28.0`.

Please test and close the issue if it works for you.

**Comment by manuelseeger** [comment]
Confirmed! Thanks a lot for the fix.
